docs/sources/alerting/set-up/provision-alerting-resources/terraform-provisioning/index.md

```diff
@@ -0,0 +1,316 @@
+
+This fragment can then be managed separately in Terraform:
+
+```terraform
+resource "grafana_message_template" "my_alert_template" {
+    name = "Alert Instance Template"
+
+    template = <<EOT
+{{ define "Alert Instance Template" }}
+Firing: {{ .Labels.alertname }}
+Silence: {{ .SilenceURL }}
+{{ end }}
+EOT
+}
+```
+
+## Provision notification policies and routing
+
+Notification policies tell Grafana how to route alert instances, as opposed to where. They connect firing alerts to your previously defined contact points using a system of labels and matchers.
+
+To provision notification policies and routing, complete the following steps.
+
+1. Copy this code block into a .tf file on your local machine.
+
+In this example, the alerts are grouped by `alertname`, which means that any notifications coming from alerts which share the same name, are grouped into the same Slack message.
+
+resource "grafana_notification_policy" "my_policy" {
+group_by = ["alertname"]
+contact_point = grafana_contact_point.my_slack_contact_point.name
+
+    group_wait = "45s"
+    group_interval = "6m"
+    repeat_interval = "3h"
+
+    policy {
+        matcher {
+            label = "a"
+            match = "="
+            value = "b"
+        }
+        group_by = ["..."]
+        contact_point = grafana_contact_point.a_different_contact_point.name
+        mute_timings = [grafana_mute_timing.my_mute_timing.name]
+
+        policy {
+            matcher {
+                label = "sublabel"
+                match = "="
+                value = "subvalue"
+            }
+            contact_point = grafana_contact_point.a_third_contact_point.name
+            group_by = ["..."]
+        }
+    }
+
+}
+
+1. In the mute_timings field, link a mute timing to your notification policy.
+
+1. Run the command ‘terraform apply’.
+
+1. Go to the Grafana UI and check the details of your notification policy.
+
+**Note:**
+
+You cannot edit resources provisioned from Terraform from the UI. This ensures that your alerting stack always stays in sync with your code.
+
+1. Click **Test** to verify that the notification point is working correctly.
+
+## Provision mute timings
+
+Mute timings provide the ability to mute alert notifications for defined time periods.
+
+To provision mute timings, complete the following steps.
+
+1. Copy this code block into a .tf file on your local machine.
+
+In this example, alert notifications are muted on weekends.
+
+resource "grafana_mute_timing" "my_mute_timing" {
+name = "My Mute Timing"
+
+    intervals {
+        times {
+          start = "04:56"
+          end = "14:17"
+        }
+        weekdays = ["saturday", "sunday", "tuesday:thursday"]
+        months = ["january:march", "12"]
+        years = ["2025:2027"]
+    }
+
+}
+
+1. Run the command ‘terraform apply’.
+1. Go to the Grafana UI and check the details of your mute timing.
```

Comment:
Mute timings don't have an effect on their own. They need to be referenced by a notification policy, in order to apply them to actual alerts. We should add that as a step here after "Go to the Grafana UI", maybe something like:

```
1. Reference your newly created mute timing in a notification policy using the `mute_timings` field. This will apply your mute timing to some or all of your notifications.
```